src/loan/loan_request.ts

```diff
@@ -292,15 +301,40 @@ export class LoanRequest extends Agreement {
         return expirationTimestamp.lt(approximateNextBlockTime);
     }
 
+    /**
+     * Returns whether the loan request has been signed by a debtor.
+     *
+     * @example
+     * loanRequest.isSignedByDebtor();
+     * => true
+     *
+     * @return {boolean}
+     */
     public isSignedByDebtor(): boolean {
         return this.data.debtorSignature !== NULL_ECDSA_SIGNATURE;
     }
 
-    public async signAsDebtor() {
+    /**
+     * Eventually signs the loan request as the debtor.
+     *
+     * @throws Throws if the loan request is already signed by a debtor.
+     *
+     * @example
+     * loanRequest.signAsDebtor();
+     * => Promise<void>
+     *
+     * @return {void}
+     */
+    public async signAsDebtor(debtorAddress?: string): Promise<void> {
         if (this.isSignedByDebtor()) {
-            return;
+            throw Error(LOAN_REQUEST_ERRORS.ALREADY_SIGNED_BY_DEBTOR);
         }
 
+        this.data.debtor = await EthereumAddress.validAddressOrCurrentUser(
+            this.dharma,
+            debtorAddress,
+        );
+
         this.data.debtorSignature = this.dharma.web3.currentProvider.isMetaMask
```

Comment:
`await this.dharma.sign.asDebtor(this.data, this.dharma.web3.currentProvider.isMetaMask)`